### ISSUE TITLE
[E2E][DIND] EmptyDir tests fail.

### ISSUE THREAD

**Issue by jayunit100** [openshift/origin#6589]
Looks like a high proportion of EmptyDir tests fail on DIND.  This could be related to why #6577 fails as well.

To reproduce (using #6541):

```
make e2e ; <path-to-output-binaries>/e2e.test --provider=local --kubeconfig=/etc/origin/master/openshift-master.kubeconfig --ginkgo.focus="EmptyDir"
```
On DIND in a single VM cluster spun up w/ Vagrant.

```
Summarizing 10 Failures:

[Fail] EmptyDir volumes [It] volume on tmpfs should have the correct mode using FSGroup [Conformance] [Skipped]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1168

[Fail] EmptyDir volumes [It] new files should be created with FSGroup ownership when container is non-root [Conformance] [Skipped]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1123

[Fail] EmptyDir volumes [It] should support (non-root,0666,tmpfs) [Conformance]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1168

[Fail] EmptyDir volumes [It] should support (non-root,0644,tmpfs) [Conformance]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1168

[Fail] EmptyDir volumes [It] should support (root,0644,tmpfs) [Conformance]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1168

[Fail] EmptyDir volumes [It] new files should be created with FSGroup ownership when container is root [Conformance] [Skipped]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1123

[Fail] EmptyDir volumes [It] should support (root,0777,tmpfs) [Conformance]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1168

[Fail] EmptyDir volumes [It] volume on tmpfs should have the correct mode [Conformance]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1168

[Fail] EmptyDir volumes [It] should support (root,0666,tmpfs) [Conformance]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1168

[Fail] EmptyDir volumes [It] should support (non-root,0777,tmpfs) [Conformance]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1168

Ran 18 of 204 Specs in 234.453 seconds
FAIL! -- 8 Passed | 10 Failed | 0 Pending | 186 Skipped --- FAIL: TestE2EWrapper (234.56s)
```

Since several of the E2Es in general fail on openshift, I validated this on on VMs in a ansible based OpenShift cluster created for me by mister @timothysc.  Good news is that most of these tests **should** pass on a regular openshift cluster.

```
Summarizing 2 Failures:

[Fail] EmptyDir volumes [It] new files should be created with FSGroup ownership when container is non-root [Conformance] [Skipped]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1123

[Fail] EmptyDir volumes [It] new files should be created with FSGroup ownership when container is root [Conformance] [Skipped]
/vagrant/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e/util.go:1123

Ran 18 of 204 Specs in 1067.768 seconds
FAIL! -- 16 Passed | 2 Failed | 0 Pending | 186 Skipped --- FAIL: TestE2EWrapper (1067.80s)
FAIL
```

**Comment by jayunit100** [comment]
cc @marun I think this might uncover some the issues with mounting certs in dind ?

**Comment by marun** [comment]
@jayunit100 I'm able to reproduce with the networking suite as well (command assumes 'hack/dind-cluster.sh start' has already been run):

NETWORKING_E2E_FOCUS="EmptyDir" OPENSHIFT_CONFIG_ROOT=dind test/extended/networking.sh

I'll see if the vm-based devcluster has the same failures as dind.

**Comment by marun** [comment]
I can confirm that a vm devcluster has the same 2 failures you report, so the fault is definitely with dind.

**Comment by jayunit100** [comment]
Asweoms, thanks for that.   Here is a spreadsheed comparing E2E results for DND for regular Openshift.  All in all about 10 failures in DIND happen that dont happen in VMs.  

**So, I think the only major difference from DIND and VMs, is in  EmptyDir functionality**

I think That is really pretty impressive overall, for DIND.

**Comment by marun** [comment]
The dind test failures appear to be due to EmptyDir volumes not being mounted as tmpfs within dind pods.  

For a pod with EmptyDir foo-storage mounted at /foo, here are the results of running `mount | grep foo-storage`...

...on the pod's node:

    tmpfs on /var/lib/openshift.local.volumes/pods/f40da62f-dabd-11e5-a2e9-5254008c3894/volumes/kubernetes.io~empty-dir/foo-storage type tmpfs (rw,relatime,rootcontext="system_u:object_r:svirt_sandbox_file_t:s0:c0,c1",seclabel)

...on the dind pod's node:

    tmpfs on /var/lib/openshift.local.volumes/pods/35e280be-dabf-11e5-b0ad-024255cba583/volumes/kubernetes.io~empty-dir/foo-storage type tmpfs (rw,relatime,seclabel)

No real difference here.  The pods do show a difference though...

...in the pod:

    tmpfs on /foo type tmpfs (rw,relatime,rootcontext="system_u:object_r:svirt_sandbox_file_t:s0:c0,c1",seclabel)

...in the dind pod:

    /dev/mapper/docker-252:1-524426-7c8733a90cde1118c3692e973acb774702493e009be4437611cd215f71489912 on /foo type xfs (rw,relatime,seclabel,nouuid,attr2,inode64,sunit=1024,swidth=1024,noquota)
 
foo-storage is created as tmpfs on both the dind and non-dind nodes, but /foo is mounted as the underlying filesystem (xfs) in the dind pod and tmpfs in the non-dind pod.  Since the kube e2e tests check explicitly for the mount type of EmptyDir volumes, this discrepancy is the cause of the failing EmptyDir tests.

**Comment by marun** [comment]
A further discovery: The different mount types between the dind node and pod are reflective of the pod and node not sharing the same EmptyDir volume, which prevents information sharing via secrets or the downward api.

**Comment by marun** [comment]
Fixed by #7668
